README.md

```diff
@@ -1 +1,420 @@
+
+### 2.1.1 Caching & Invalidation
+
+Signing CIDs has two additional caching effects
+
+* Signing CIDs enables a simple strategy for caching validation by CID
+* Such a strategy also requires accounting for revocation of the signing keys, and so need to be tagged with this additional metadata
+
+## 2.2 Raw (Noncanonicalized) Data
+
+Canonicalization is not required if data is encoded as raw bytes (multicodec `%x55`). The exact bytes are already present, and MUST not be changed.
+
+# 3 Varsig Format
+
+After being decoded from [unsigned varint]s, a varsig includes the following segments:
+
+```abnf
+varsig = multibase-prefix %x34 varsig-header varsig-body
+multibase-prefix = ALPHA ; Multibase
+varsig-header = unsigned-varint ; Usually the public key code from Multicodec
+varsig-body = *OCTET; Zero or more segments required by the kind of varsig (e.g. raw bytes, hash algorithm, etc)
+```
+
+For example, here is an EdDSA signature for some content encoded as DAG-PB:
+
+`%x34ed01ae3784f03f9ee1163382fa6efa73b0c31ecf58c899c836709303ba4621d1e6df20e09aaa568914290b7ea124f5b38e70b9b69c7de0d216880eac885edd41c302`
+
+### 3.1 Varsig Prefix
+
+The varsig prefix MUST be `%x34`.
+
+### 3.2 Signature Header
+
+The prefix of the signature algorithm. This is often the [multicodec] of the associated public key, but MAY be unique for the signature type. The code MAY live outside the multicodec table. This field MUST act as a discriminant for how many expected fields come in the varsig body, and what each of them mean.
+
+### 3.3 Varsig Body
+
+The varsig body MUST consist of one or more segments, and MUST be defined by the signature algorithm.
+
+Some examples include:
+
+* Raw signature bytes only
+* CID of [DKIM] certification transparency record, and raw signature bytes
+* Hash algorithm multicodec prefix, data encoding prefix, signature counter, nonce, HMAC, and raw signature bytes
+
+# 4 Payload Encoding
+
+The [IPLD] data model is encoding agnostic by design. This is very convenient in many applications, such as making for very convenient conversions between types for transmission versus encoding. Unfortunately signatures require signing over specific bytes, and thus over a specific encoding of the data.
+
+To facilitate this, the type `encoding-info` MAY be used:
+
+``` abnf
+encoding-info
+  = %x55   ; Raw bytes multicodec prefix
+  / %x70   ; DAG-PB multicodec prefix
+  / %x71   ; DAG-CBOR multicodec prefix
+  / %x0129 ; DAG-JSON multicodec prefix
+  / %x6A77 ; JWT
+  / %xE191 eip-191-body encoding-info ; EIP-191
+  
+eip-191-body 
+  = %x00 eth-validator-address 
+  / %x01 
+  / %x45 message-byte-length ; versions & payloads
+
+eth-validator-address = 20(OCTET)
+message-byte-length = unsigned-varint
```

Comment:
I should probably point you directly at it @oed 😉

Reply:
I don't think we should include the ethereum address that made the signature here. Will be redundant in many cases. The only version of eip191 used in practice is 0x45 so I would suggest that's what we use.

I think the encoding info for a eip191 msg should look like this:
`encoding-info = varint(0xe191) varint(codec)`
where codec could be dag-cbor, dag-json, or something custom, etc.

Reply:
> The only version of eip191 used in practice is 0x45 so I would suggest that's what we use.

Any objection to keeping the version in there, just dropping the other two?

```abnf
encoding-info
  = %x70   ; DAG-PB multicodec prefix
  ;; ...
  / %xE191 %x45 message-byte-length encoding-info ; EIP-191 version 0x45
```

Reply:
Actually, you're absolutley right that we probably won't see `0x00`, and `0x01` is also known as EIP-712, so it could have its own row. Okay that makes sense, I'll remove the `%x45` 👍